slack_sdk/oauth/installation_store/amazon_s3/__init__.py

```diff
@@ -282,15 +283,52 @@ def delete_installation(
             Bucket=self.bucket_name,
             Prefix=f"{workspace_path}/installer-{user_id or ''}",
         )
+        deleted_keys = []
         for content in objects.get("Contents", []):
             key = content.get("Key")
             if key is not None:
                 self.logger.info(f"Going to delete installation ({key})")
                 try:
                     self.s3_client.delete_object(
                         Bucket=self.bucket_name,
-                        Key=content.get("Key"),
+                        Key=key,
                     )
+                    deleted_keys.append(key)
+                except Exception as e:  # skipcq: PYL-W0703
+                    message = f"Failed to find bot installation data for enterprise: {e_id}, team: {t_id}: {e}"
+                    raise SlackClientConfigurationError(message)
+
+                try:
+                    no_user_id_key = key.replace(f"-{user_id}", "")
+                    if not no_user_id_key.endswith("installer-latest"):
+                        self.s3_client.delete_object(
+                            Bucket=self.bucket_name,
+                            Key=no_user_id_key,
+                        )
+                        deleted_keys.append(no_user_id_key)
                 except Exception as e:  # skipcq: PYL-W0703
                     message = f"Failed to find bot installation data for enterprise: {e_id}, team: {t_id}: {e}"
-                    self.logger.warning(message)
+                    raise SlackClientConfigurationError(message)
+
+        # Check the remaining installation data
+        objects = self.s3_client.list_objects(
+            Bucket=self.bucket_name,
+            Prefix=f"{workspace_path}/installer-",
+            MaxKeys=10,  # the small number would be enough for this purpose
+        )
+        keys = [
+            c.get("Key")
+            for c in objects.get("Contents", [])
+            if c.get("Key") not in deleted_keys
```

Comment:
As S3 storage is now strongly consistent, we may not need to remove these keys, though. https://aws.amazon.com/blogs/aws/amazon-s3-update-strong-read-after-write-consistency/